instant_ngp_3dml/density.py

```diff
@@ -1,23 +1,41 @@
 #!/usr/bin/env python3
 """Density Extraction Script"""
+import math
 import os
-from instant_ngp_3dml.utils.io import read_json
-import numpy as np
 
+import numpy as np
 import pyngp as ngp  # noqa
 
+from instant_ngp_3dml.utils.io import read_json
+from instant_ngp_3dml.utils.io import write_json
 from instant_ngp_3dml.utils.log import logger
 from instant_ngp_3dml.utils.profiler import profile
 
 
 @profile
-def density(snapshot_msgpack: str,
-            nerf_config_file: str,
-            output_folder: str,
-            resolution: int = 256,
-            thresh: float = 2.5,
-            density_range: float = 4.0):
-    """Extract Density from NeRF Volume"""
+def main(snapshot_msgpack: str,
+         nerf_config_json: str,
+         out_density_folder: str,
+         resolution: int = 256,
+         thresh: float = 2.5,
+         density_range: float = 4.0,
+         cube_size: int = 4,
+         as_binary: bool = False):
+    """Extract Density from NeRF Volume.
+
+        The NeRF volume is divided into cubes where each cube has two files:
+        the 3d grid and a json with the grid bounding box.
+
+        Args:
+            snapshot_msgpack: Input NeRF Weight
+            nerf_config_json: Input NeRF Configuration Json
+            out_density_folder: Output folder with density images
+            resolution: Grid resolution in voxel by meters
+            thresh: Treshold value for density extraction
+            density_range: Density Range to map density value in grayscale
+            cube_size: Size of extracted cube,
```

Comment:
La resolution est deja en parametres d'entrée
resolution: Grid resolution in voxel by meters

Reply:
La resolution est un entier qui vaut 256
L'utilisateur a plus souvent envie de dire "precision=1cm", d'où cube_size s'en déduit en prenant resolution*1cm

Reply:
precision_m: float
grid_resolution: int

Reply:
precision = cube_size / grid_resolution